iModelCore/iModelPlatform/DgnCore/linestyle/LsDb.cpp

```diff
@@ -659,29 +659,28 @@ DgnElementPtr LineStyleElement::_CloneForImport(DgnDbStatus* status, DgnModelR d
     if(!context.IsBetweenDbs())
         return newElem;
 
+    Json::Value  jsonObj (Json::objectValue);
+
+    Utf8String  srcData (this->GetData());
+    if (!Json::Reader::Parse(srcData, jsonObj))
+      return newElem;
+    LsComponentId compId = LsDefinition::GetComponentId (jsonObj);
+    
     DgnStyleId dstStyleId = QueryId(context.GetDestinationDb(), this->GetName().c_str());
     if (dstStyleId.IsValid())
         {
-        //  *** TBD: Check if the line style definitions match. If not, rename and remap
-        context.AddLineStyleId(DgnStyleId(this->GetElementId().GetValue()), dstStyleId);
-        DgnElementId elemId = context.AddElementId(this->GetElementId(), DgnElementId(dstStyleId.GetValue()));
-        
-        return context.GetDestinationDb().Elements().GetForEdit<LineStyleElement>(elemId);
-        }
+        LineStyleElementPtr dstElem = context.GetDestinationDb().Elements().GetForEdit<LineStyleElement>(dstStyleId);

```

Comment:
Should not be using GetForEdit, since you aren't going to change this element. Should also add a comment to explain why this is necessary.